src/main/java/com/nvs/store/service/ProductService.java

```diff
@@ -27,12 +28,13 @@ public Product addProduct(Product product) {
     }
 
     public Product updateProduct(Long id, Product product) {
-        Product updProduct = productRepository.getProductById(id);
+        Product updProduct = productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotExistsException(id));
         updProduct.setTitle(product.getTitle());
         updProduct.setAvailable(product.getAvailable());
         updProduct.setPrice(product.getPrice());
         productRepository.save(updProduct);
-        return productRepository.getProductById(id);
+        return updProduct;
```

Comment:
Perhaps here you could return the return value of the save method, which would be the object saved into the DB.